### DISCUSSION TITLE
Support to add tiktok videos

### DISCUSSION THREAD

**Discussion by lerra** [mealie-recipes/mealie#1474] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x] Reviewed the existing feature requests and my feature has not been requested.
- [x] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Sometimes, my friends sends me a recommendation on a meal recipe from tiktok and I want to add it to my "todo" of testing the meal. I can not import a tiktok video recipe via a URL, an error (could not be found) is given.

**(Optional): Suggest A Solution**
The title from the video could be taken and the link or video embedded on the page and leave the rest unfilled.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by usma0118** [comment]
+1 
Add youtube and instagram to list.

**Comment by michael-genson** [comment]
This is implemented with the fallback opengraph scraper. As long as the URL has opengraph tags, it will pull the image, title, description, etc.

**Comment by JoTec2002** [comment]
I wanted to have a similar feature. Mealie already offers some integration with OpenAI/Ollama, but I wasn't fully satisfied with the results.

It doesn't support TikTok but Instagram at least.

So, I created a small Docker container that explicitly scrapes Instagram posts using an Instagram scraper and allows Mealie to create a new recipe based on the video description. Afterward, it uploads the Instagram image and video directly to Mealie.

I hope this might help someone: https://github.com/JoTec2002/InstagramToMealie/
It creates a small Website were you can input the Instagram URL. I have it deployed in the same Docker-Compose file as Mealie.

I hope it’s okay to share this here—this isn’t for marketing; I just thought it might be useful to someone else!

**Comment by stanthewizzard** [comment]
tik tok is a must
